clang/test/Sema/format-strings-ms.c

```diff
@@ -85,4 +86,31 @@ void z_test(void *p) {
   scanf("%Z", p); // expected-warning{{invalid conversion specifier 'Z'}}
 }
 
+void w_int_test(void) {
```

Comment:
1. I can't find any tests to verify the print result is correct. Do I miss something?
2. One more test about error handling: `unsupported size` test is missing. Like `printf("%w20d")` should be thrown or raise errors. Should I raise error or any methods to throw the error?

Reply:
The printed result will depend on the C standard library used, so there's no way to verify that here. As for unsupported sizes, I think those should be emitted as a warning; you'll likely need to add a new warning for this along the lines of "format specifies %select{an exact-|a fastest-}0width integer type with invalid bit-width %1".